src/components/BarChartCard/BarChartCard.jsx

```diff
@@ -3,212 +3,101 @@ import SimpleBarChart from '@carbon/charts-react/bar-chart-simple';
- * @param {Array<string>} datasetNames unique dataset bar names to be used if color is an object
- *
- * @returns {Object} colors - formatted
- */
-export const formatColors = (series, datasetNames) => {
-  // first set the carbon charts config defaults
-  const colors = { identifier: 'group', scale: {} };
-  // if color is an array, order doesn't matter so just map as many as possible
-  if (Array.isArray(series[0].color)) {
-    series[0].color.forEach((color, index) => {
-      colors.scale[datasetNames[index]] = color;
-    });
-  } else {
-    series.forEach(dataset => {
-      if (dataset.color) {
-        // if its a string, set the color to this line
-        if (typeof dataset.color === 'string') {
-          colors.scale[dataset.label] = dataset.color;
-        } // If its an object, use it, but the keys must match the labels provided from series.label
-        else {
-          colors.scale = dataset.color;
-        }
-      }
-    });
-  }
-
-  return colors;
-};
+const memoizedGenerateSampleValues = memoize(generateSampleValues);
 
 const BarChartCard = ({
-  title,
-  content: {
-    layout = BAR_CHART_LAYOUTS.VERTICAL,
-    type = BAR_CHART_TYPES.SIMPLE,
-    xLabel,
-    yLabel,
-    series,
-    categoryDataSourceId,
-    timeDataSourceId,
-  },
+  title: titleProp,
+  content,
   size,
-  values,
+  values: initialValues,
   locale,
   i18n,
   isExpanded,
   isLazyLoading,
+  isEditable,
+  isLoading,
+  interval,
   className,
   ...others
 }) => {
+  const { noDataLabel } = i18n;
+  const {
+    title,
+    content: {
+      series,
+      timeDataSourceId,
+      categoryDataSourceId,
+      layout = BAR_CHART_LAYOUTS.VERTICAL,
+      xLabel,
+      yLabel,
+      unit,
+      type,
+    },
+    values: valuesProp,
+  } = handleCardVariables(titleProp, content, initialValues, others);
+
+  // Check for unique prop type errors. The conditionals must pass false to throw the prop error
+  // all charts must have oneOf[categoryDataSourceId, timeDataSourceId]
+  warning(
+    !(!categoryDataSourceId && !timeDataSourceId),
+    `\`BarChartCard\` must have \`categoryDataSourceId\` OR \`timeDataSourceId\`.`
+  );
+  // GROUPED charts can't have timeDataSourceId
+  warning(
+    !(type === BAR_CHART_TYPES.GROUPED && timeDataSourceId),
+    `\`BarChartCard\` of type \`GROUPED\` cannot use \`timeDataSourceId\`.`
+  );
+  // STACKED charts with timeDataSourceId and categoryDataSourceId can't have datasource labels
+  if (type === BAR_CHART_TYPES.STACKED && timeDataSourceId && categoryDataSourceId) {
+    let shouldNotGivePropError = true;
+    series.forEach(datasource => {
+      if (datasource.label) {
+        shouldNotGivePropError = false;
+      }
+    });
+    warning(
+      shouldNotGivePropError,
+      `\`BarChartCard\` of type \`STACKED\` with \`categoryDataSourceId\` AND \`timeDataSourceId\` cannot use \`label\` within series. The labels will be created from the \`categoryDataSourceId\`.`
+    );
+  }
```

Comment:
Could these also be custom proptype validation functions, like the additions you made to the CardPropTypes? We wouldn't want to pop these warnings in production.

Reply:
Yep, thought there was some limitations, but there's actually not. Fixing them